### DISCUSSION TITLE
PlayScheduled for MIDI files

### DISCUSSION THREAD

**Discussion by LeoMassaFM** [melanchall/drywetmidi#266] [Q&A] [answered]
First of all, I just want to say that I'm impressed with how versatile and useful this plugin is, even if it's a little cumbersome to use at first.
I've been looking for answers for a really long time, but never found any, nor people discussing about it, so here's my question: is there a way to make a script read a MIDI file only after a certain amount of time has passed, similar to what PlayScheduled does?
For instance, when I want to play an AudioSource or AudioClip after 3 seconds, I type "myAudioSource.PlayScheduled(AudioSettings.dspTime + 3)". Is it possible to achieve the same effect but with a MIDI file? Thanks.

**Comment by melanchall** [comment, marked as answer]
Hi,

First of all, unfortunately I have no idea what PlayScheduled is. If you want to read a MIDI file after some time, just use standard .NET timer. When a timer fires, call `MidiFile.Read`.

**Reply by melanchall** [reply]
@LeoMassaFM Any news?

**Reply by LeoMassaFM** [reply]
No, but I'll see if I can figure it out on my own. Thanks.
